enhancements/microshift/etcd-as-transient-systemd-unit.md

```diff
@@ -0,0 +1,123 @@
+---
+title: etcd
+authors:
+  - dusk125
+  - hasbro17
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - TBD
+approvers:
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# etcd as a Transient systemd Unit
+
+## Summary
+
+The enhancement proposes moving etcd out of the MicroShift binary and into a transient systemd unit that is launched and managed by MicroShift.
+
+## Motivation
+
+Currently MicroShift [bundles upstream etcd](https://github.com/openshift/microshift/blob/f7f2260e4fbff61654f478fc149cb7052261f87f/go.mod#L21) into its binary and runs it as a goroutine. This results in etcd dependencies being coupled with MicroShift which makes it harder to maintain and upgrade etcd independently of the platform like it is done with a separate openshift/etcd repo in the Openshift Container Platform (OCP).
+
+This enhancement outlines an update to MicroShift that would decouple its deployment of etcd to bring it closer to how etcd is maintained with OCP.
```

Comment:
I agree

Reply:
> The goal is not to deploy etcd more closely to how OCP deploys it; depending on the chosen deployment method this would be a welcome side-effect.

It may not be a goal for you, but the etcd team maintains this long term and the maintenance and support burdens are significantly reduced by having more closely aligned deployments.  It seems like a reasonable goal for the long term maintainers and supporters of this product and code.

Reply:
> > The goal is not to deploy etcd more closely to how OCP deploys it; depending on the chosen deployment method this would be a welcome side-effect.
> 
> It may not be a goal for you, but the etcd team maintains this long term and the maintenance and support burdens are significantly reduced by having more closely aligned deployments. It seems like a reasonable goal for the long term maintainers and supporters of this product and code.

If the goal of aligning etcd maintenance drives away the end goal of the whole project, there is no point on doing anything like this. And this not only applies to etcd, but to the whole project.